library/src/CMakeLists.txt

```diff
@@ -76,6 +78,7 @@ if(HIP_PLATFORM STREQUAL amd)
     else()
       find_package( rocblas REQUIRED CONFIG PATHS /opt/rocm /opt/rocm/rocblas )
     endif( )
+    list(APPEND static_depends PACKAGE rocblas)
   endif( )
 
   target_link_libraries( hipblas PRIVATE roc::rocblas hip::host )
```

Comment:
> Probably in wrong scope, I would suggest move it to right above target_link_libraries line for consistency.

The things that go into `static_depends` are package names.  I believe this is in the correct scope, because all of the logic that would find the `rocblas` package is contained in this scope.  I'm not even sure when `TARGET rocblas` might ever be true, but if it was, then `rocblas` would not be a separate package that needs to go into `static_depends`.

> Will need hip::host as well in this list.
> But are namespace targets not allowed ? Could create rocm_link_libraries that adds to both?

Again, these are packages, not libraries or targets.  Since `hip` is already added to non-static `DEPENDS`, and `DEPENDS` is a stronger linkage than `STATIC_DEPENDS`, we should already be doing what you're looking for.

Reply:
I meant inside the if( NOT TARGET rocblas ) scope, not referring here to the namespace scope.  Any thing that finds it earlier would set the target, don't think we should assume you are the first location to find something or too brittle.    So if rocblas was already found that isn't relevant it is a dependency if you link, right?  

maybe it should be in DEPENDS like rocSOLVER does, as if I understand you correctly,  static depends is only for libs not in DEPENDS.   dynamic_depends could be created for solver, and go in DEPENDS section.

Reply:
Moved the scope of the APPEND as suggested.  Dynamic depends aren't necessary, as downstream CMake usage of hipBLAS does not require a `find_package` for any of these libraries.